contracts/schemes/JoinAndQuit.sol

```diff
@@ -0,0 +1,243 @@
+pragma solidity ^0.5.16;
+
+import "../votingMachines/VotingMachineCallbacks.sol";
+import "@openzeppelin/upgrades/contracts/Initializable.sol";
+import "../libs/StringUtil.sol";
+
+
+/**
+ * @title A scheme for join in a dao.
+ * - A member can be proposed to join in by sending a min amount of fee.
+ * - A member can ask to quite (RageQuit) a dao on any time.
+ * - A member can donate to a dao.
+ */
+contract JoinAndQuit is
+        VotingMachineCallbacks,
+        ProposalExecuteInterface,
+        Initializable {
+    using SafeMath for uint;
+    using SafeERC20 for address;
+    using StringUtil for string;
+
+    event JoinInProposal(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        string _descriptionHash,
+        address _proposedMember,
+        uint256 _fundAmount
+    );
+
+    event FundedDeadLineReached(
+        address indexed _avatar
+    );
+
+    event Donation(
+        address indexed _avatar,
+        uint256 indexed _donation
+    );
+
+    event RageQuit(
+        address indexed _avatar,
+        uint256 indexed _refund
+    );
+
+    event RedeemReputation(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address indexed _beneficiary,
+        uint256 _amount);
+
+    event ProposalExecuted(address indexed _avatar, bytes32 indexed _proposalId, int256 _decision);
+
+    struct Proposal {
+        bool accepted;
+        address proposedMember;
+        address funder;
+        uint256 funding;
+    }
+
+    mapping(bytes32=>Proposal) public proposals;
+    mapping(address=>uint256) public fundings;
+
+    IntVoteInterface public votingMachine;
+    bytes32 public voteParams;
+    Avatar public avatar;
+    IERC20 public fundingToken;
+    uint256 public minFeeToJoin;
+    uint256 public memberReputation;
+    uint256 public fundingGoal;
+    uint256 public fundingGoalDeadLine;
+    uint256 public totalDonation;
+    string public constant FUNDED_BEFORE_DEADLINE_KEY = "FUNDED_BEFORE_DEADLINE";
+    string public constant FUNDED_BEFORE_DEADLINE_VALUE = "TRUE";
+
+    /**
+     * @dev initialize
+     * @param _avatar the avatar this scheme referring to.
+     * @param _votingMachine the voting machines address to
+     * @param _voteParams voting machine parameters.
+     * @param _fundingToken the funding token
+     * @param _minFeeToJoin minimum fee required to join
+     * @param _memberReputation the repution which will be allocated for members
+     * @param _fundingGoal the funding goal
+     * @param _fundingGoalDeadLine the funding goal deadline
+     */
+    function initialize(
+        Avatar _avatar,
+        IntVoteInterface _votingMachine,
+        bytes32 _voteParams,
+        IERC20 _fundingToken,
```

Comment:
This means we will not be able to fund with ETH? 

I am not sure if XDAI on the poa network is an ERC20 token (I believe not, but I'll need to check)

Reply:
`This means we will not be able to fund with ETH?`
Is that a requirement ? I never saw that in the spec .

- xdai is erc677 which is erc20 compatible

Reply:
Funding with ETH is not particularly a requirement (it would be nice, but we do not need that). But funding with XDAI is definitely something we want to do :-)

And as far as I undstand (but again, I am not 100% sure), xdai is the "native token" on the xdai network, meaning it is like ETH on the normal chain, and not an ERC20 contract...